### ISSUE TITLE
Build failure - GL_PATCHES was not declared in this scope

### ISSUE THREAD

**Issue by colemickens** [WayfireWM/wayfire#135]
I'm not sure what I'm doing wrong, but my `wayfire` package definition stopped building recently:

```
[139/155] Compiling C++ object 'plugins/cube/643b030@@cube@sha/cube.cpp.o'.
FAILED: plugins/cube/643b030@@cube@sha/cube.cpp.o
/nix/store/isg8rxaxkijl9x3hr2gzsf8pqfnqxg3k-gcc-wrapper-7.4.0/bin/c++ -Iplugins/cube/643b030@@cube@sha -Iplugins/cube -I../plugins/cube -I../src/api -I. -I../ -I/nix/store/ppg31x6w5ss6rslacpbyph3sl58gblhq-wlroots-36ba56140ae325ae398229a689f20ded126608bc/include -I/nix/store/l0c5zzypk9n9qwvz3n7iai415q3shy5i-pixman-0.36.0/include/pixman-1 -I/nix/store/k2nwdw4wl1q3pazlz5lw70d7p6y2hg6b-wf-config-6d3426e216ac62ffa035035f9c1bea074e184018/include/wayfire -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -Wextra -std=c++11 -O3 -DWLR_USE_UNSTABLE -Wno-unused-parameter -fPIC  -MD -MQ 'plugins/cube/643b030@@cube@sha/cube.cpp.o' -MF 'plugins/cube/643b030@@cube@sha/cube.cpp.o.d' -o 'plugins/cube/643b030@@cube@sha/cube.cpp.o' -c ../plugins/cube/cube.cpp
../plugins/cube/cube.cpp: In member function 'void wayfire_cube::load_program()':
../plugins/cube/cube.cpp:184:26: warning: unused variable 'tcs' [-Wunused-variable]
         GLuint vss, fss, tcs = -1, tes = -1, gss = -1;
                          ^~~
../plugins/cube/cube.cpp:184:36: warning: unused variable 'tes' [-Wunused-variable]
         GLuint vss, fss, tcs = -1, tes = -1, gss = -1;
                                    ^~~
../plugins/cube/cube.cpp:184:46: warning: unused variable 'gss' [-Wunused-variable]
         GLuint vss, fss, tcs = -1, tes = -1, gss = -1;
                                              ^~~
In file included from ../plugins/cube/cube.cpp:1:0:
../plugins/cube/cube.cpp: In member function 'void wayfire_cube::render_cube(GLuint)':
../plugins/cube/cube.cpp:445:40: error: 'GL_PATCHES' was not declared in this scope
                 GL_CALL(glDrawElements(GL_PATCHES, 6, GL_UNSIGNED_INT, &indexData));
                                        ^
../src/api/opengl.hpp:31:20: note: in definition of macro 'GL_CALL'
 #define GL_CALL(x) x; gl_call(__PRETTY_FUNCTION__, __LINE__, __STRING(x))
                    ^
../plugins/cube/cube.cpp:445:40: note: suggested alternative: 'GL_DITHER'
                 GL_CALL(glDrawElements(GL_PATCHES, 6, GL_UNSIGNED_INT, &indexData));
                                        ^
../src/api/opengl.hpp:31:20: note: in definition of macro 'GL_CALL'
 #define GL_CALL(x) x; gl_call(__PRETTY_FUNCTION__, __LINE__, __STRING(x))
                    ^
```

**Comment by soreau** [comment]
There were some patches pushed to make things build without gles32 but still need something like this:

```
diff --git a/plugins/cube/cube.cpp b/plugins/cube/cube.cpp
index 3ee68bf..224cb57 100644
--- a/plugins/cube/cube.cpp
+++ b/plugins/cube/cube.cpp
@@ -440,12 +440,15 @@ class wayfire_cube : public wayfire_plugin_t
 
             auto model = calculate_model_matrix(i);
             GL_CALL(glUniformMatrix4fv(program.modelID, 1, GL_FALSE, &model[0][0]));
-
+#ifdef USE_GLES32
             if (tessellation_support) {
                 GL_CALL(glDrawElements(GL_PATCHES, 6, GL_UNSIGNED_INT, &indexData));
             } else {
+#endif
                 GL_CALL(glDrawElements(GL_TRIANGLES, 6, GL_UNSIGNED_INT, &indexData));
+#ifdef USE_GLES32
             }
+#endif
         }
     }
```

**Comment by ammen99** [comment]
@soreau if that patch fixes compilation without GLES 3.2, send a PR

**Comment by soreau** [comment]
Created PR [137](https://github.com/WayfireWM/wayfire/pull/137).

**Comment by soreau** [comment]
@colemickens should be fixed with d31bff63, please reopen if it doesn't.
